syslog-ng/SOURCES/syslog-ng.init

```diff
@@ -49,13 +51,16 @@ prepare() {
   local has_errors=""
 
   [[ ! -f $binary ]] && has_errors=true && kv.error "Error! File <$binary> does not exist"
-  [[ ! -x $binary ]] && has_errors=true && kv.error "Error! File <$binary> does not executable"
+  [[ ! -x $binary ]] && has_errors=true && kv.error "Error! File <$binary> is not executable"
 
   [[ ! -f $conf_file ]] && has_errors=true && kv.error "Error! File <$conf_file> does not exist"
   [[ ! -r $conf_file ]] && has_errors=true && kv.error "Error! File <$conf_file> does not readable"
 
-  [[ ! -f $persist_file ]] && has_errors=true && kv.error "Error! File <$persist_file> does not exist"
-  [[ ! -r $persist_file ]] && has_errors=true && kv.error "Error! File <$persist_file> does not readable"
+  [[ ! -d $persist_dir ]] && has_errors=true && kv.error "Error! Directory <$persist_dir> does not exist"
+  [[ ! -w $persist_dir ]] && has_errors=true && kv.error "Error! Directory <$persist_dir> is not writable"
+
+  [[ ! -d ${kv[pid_dir]} ]] && has_errors=true && kv.error "Error! Directory <${kv[pid_dir]}> does not exist"
+  [[ ! -w ${kv[pid_dir]} ]] && has_errors=true && kv.error "Error! Directory <${kv[pid_dir]}> is not writable"
```

Comment:
This is useless checks. If `kv[pid_dir]` is not exist, kaosv will create this directory.